apps/hash-agents/pyproject.toml

```diff
@@ -38,8 +38,13 @@ target-version = ['py311']
 preview = true
 
 [tool.ruff]
-select = ["ANN", "B", "C901", "E", "F", "G201", "N", "UP", "S", "T20", "TRY"]
-ignore = ["ANN101"]
+select = ["ALL"]
```

Comment:
I am unsure if that is a good idea; for example, `COM812` directly influences the output of black and clashes with `COM819`, the `I` family of lints is already covered by `lint:isort`

Reply:
I think we can get rid of `lint:isort` then, unless you have an argument against it. The lints that clash are automatically detected by `ruff`.

Everywhere else we have tried to be about as constrained as we can with lints (as in the opposite of permissive and just enable all of them until they prove annoying), I do not see why we wouldn't do that here.

The ruff guide does say that most people probably don't want `ALL` because it means lints will be implicitly enabled when you update versions, but that's actually the same strategy we employ with `clippy`.

Reply:
Do you have an argument for isort instead of `I`? I feel like reducing the tooling down would be better, and linting will fail until we resolve which one to use.

Reply:
we need isort anyway for `fix`, but I am fine with the `I` lint instead of `lint:isory`

Reply:
Why do we need `isort` for fix? I believe sorting is fixable through `ruff` is it not?

Reply:
you are completely right, yes

Reply:
Noice, I'll test out removing `isort` tomorrow

Reply:
I've removed it. [Remove isort in favour of ruff](https://github.com/hashintel/hash/pull/2493/commits/0bc3dde4f8654c1a1e7129db6bd797c380130dc6)

I tried editing the code artificially, screwing up imports all over the place, and `fix` and `lint` seems to still work quite well